www/src/pages/headless/combobox.md

```diff
@@ -0,0 +1,432 @@
+| [[Esc]] when listbox is open                  | Closes the combobox            |
+| [[Enter]] [[Space]] when the combobox is open | Selects the active item        |
+
+## Performance
+
+The combo box component used a sepcific internal pipeline to filter and display the selection items:
+
+1) Retrieve the user's input
+2) _Debounce_
+3) Find matches
+4) _Debounce_
+5) Render matches
+
+The debouncing is in place because the above worklfow consists of two relatively expensive operations: _filtering_ and
+_rendering_.
+
+While typing, the query may be manipulated multiple times per second. In order for the filter function to run as few
+times as possible, the flow of inputs is debounced.
+
+The same goes for the actual rendering: It is by far the most expensive operation in the workflow so it is debounced to
+not be executed multiple times in a row.
+
+Adding to the above, the number of displayed items in the dropdown also has an impact on the rendering performance.
+
+::: info
+Most of the time, the default behavior should be working fine. There might be cases, however, where the implementing
+component has a consistently high/low amount of items or other niche scenarios. In those cases, the debouncing and
+other performance-related parameters can be configured via the DSL.
+:::
+
+| Property                | Type   | Default | Description                                                      |
+|:------------------------|:-------|:--------|:-----------------------------------------------------------------|
+| `maximumDisplayedItems` | `Int`  | `20`    | Maximum number of items to display                               |
+| `inputDebounceMillis`   | `Long` | `50L`   | Time to wait and debounce before the filter function is invoked  |
+| `renderDebounceMillis`  | `Long` | `50L`   | Time to wait and debounce before the filter results are rendered |
+
+See [`combobox`](#combobox) for more api information.
+
+## API
+
+### Summary / Sketch
+
+```kotlin
+combobox<T> {
+    val items: ItemsHook()
+    // params: List<T> / Flow<List<T>>
+
+    var itemFormat: (T) -> String
+
+    val value: DatabindingProperty<T>
+
+    var filterBy: FilterFunctionProperty
+    // params: (Sequence<T>, String) -> Sequence<T> / T.() -> String
+
+    val selectionStrategy: SelectionStrategyProperty
+    // methods: autoSelectMatches() / manual()
+
+    var maximumDisplayedItems: Int = 20
+    var inputDebounceMillis: Long = 50L
+    var renderDebounceMillis: Long = 50L
+
+    comboboxPanelReference() { }
+    comboboxInput() { }
+    comboboxLabel() { }
+    comboboxItems() {
+        // inherited by `PopUpPanel`
+        var placement: Placement
+        var strategy: Strategy
+        var flip: Boolean
+        var skidding: Int
+        var distance: int
+
+        val results: Flow<QueryResult.ItemList<T>>
+
+        // state.render {
+        // for each QueryResult.ItemList<T>.Item<T> {
+        comboboxItem(Item<T>) { }
```

Comment:
Would it be better to add some indentation to emphasize the structure? (Even though the surrounding code is "comment"!)

We did that in `datacollection` too and imho that looks better.